### ISSUE TITLE
payment_method_nonce hidden input added twice by setting up Paypal again after teardown

### ISSUE THREAD

**Issue by cityreader** [braintree/braintree-web#73]
Hi,

We have used hosted field and paypal as separate payment methods in `Drupal Commerce`. There is payment option to let user choose credit card (`Hosted Fields`) or `PayPal`. It defaults to `PayPal`. If user change payment method, the process will be:
1 the current integration will be destroyed by using teardown method
2 payment form will be reloaded via Ajax
3 new payment method will be setup again.

After implementing this we have found `PayPal` does not work every time. It does work in the first time when page is loaded. However when user changes payment method to credit card (`Hosted Fields`) and changes back to `PayPal` again, it won't work. 

After debugging, it turns out in the second time to set up PayPal, hidden element `payment_method_nonce` is added twice. You can see this in the images below:

**First time setup PayPal**: there is one `payment_method_nonce` element added
![first time setup paypal](https://cloud.githubusercontent.com/assets/343948/10035040/5a78892c-61d9-11e5-8602-1062e89adbf8.png)

**Set up PayPal again after teardown**: there are two `payment_method_nonce` element added
![set up paypal again after teardown](https://cloud.githubusercontent.com/assets/343948/10035039/5a75f57c-61d9-11e5-8882-0fdeabbdb791.png)

This duplicate `payment_method_nonce` will cause issue. When I add my PayPal account in the popup modal, the nonce will only be added to the first `payment_method_nonce`. And the second is empty. So when I click submit to post form the second empty `payment_method_nonce` element which is after the first one in HTML will override the value of the first one. And this causes empty nonce issue. 

I cannot simply remove the second `payment_method_nonce` by JavaScript as it seems also used in Braintree. If I delete it by JS, error will be raised. As an ad hoc workaround, I copy the nonce value from the first element to the second element in the nonce received callback. However this still does not work after setuping and tearing down payment method too many times without reloading page.

Any advises? Thanks.

**Comment by EvanHahn** [comment]
@cityreader We're looking into this.

You have PayPal and Hosted Fields on your page before calling `teardown`. In this situation, do you call `braintree.setup` one time, or twice (once per payment method)?

**Comment by cityreader** [comment]
Hi @EvanHahn, in short we call `braintree.setup` method only once. We bootstrap either `PayPal` or `Hosted Fields` depending on user's' selection. PayPal is the default payment method.

**Comment by EvanHahn** [comment]
We took a look at this and couldn't reproduce your issue. If you could show us more of your integration, that would help a lot. Specifically, could we see your call(s) to `braintree.setup`?

**Comment by cityreader** [comment]
@EvanHahn sure, could you give me an email we can send raw code to you please?

**Comment by mrak** [comment]
@cityreader you can email our support team (they are standing by for issues like this) with your integration details and problem at support@braintreepayments.com

**Comment by kyledetella** [comment]
@cityreader I am going to close this, feel free to re-open if your issue is not resolved via our support channels (but I am confident they will take care of you).

**Comment by tschiman** [comment]
It would really be nice to have a posted solution here. I am facing the exact same situation as the OP.

**Comment by tschiman** [comment]
Since version 2.7.0, the configuration object passed during braintree.setup has provided an onReadycallback. As of this latest build the onReady callback will provide an integration object on which you can perform actions such as teardown. A basic example of this looks like:

var myIntegration;  

braintree.setup(TOKEN, 'dropin', {
   onReady: function (integration) {
     myIntegration = integration;
   }
 });
  // When you're ready to destroy your integration 
myIntegration.teardown(function () {
   // braintree.setup can safely be run again!
 });

Basicaly write the integration object to an external variable. When you want to start a new instance of braintree get that object and call tear down. Pass the tear down method a call back that will rebuild your braintree payment stuff.
